worker/dependency/testing/stub.go

```diff
@@ -4,30 +4,39 @@
 package testing
 
 import (
+	"fmt"
 	"reflect"
+	"strings"
 
 	"github.com/juju/errors"
 
 	"github.com/juju/juju/worker/dependency"
 )
 
-// StubResource is used to define the behaviour of a StubGetResource func for a
-// particular resource name.
+// StubResource is used to define the behaviour of a StubGetResource
+// func for a particular resource name. If both Output and Outputs are
+// set, Output takes precedence.
 type StubResource struct {
-	Output interface{}
-	Error  error
+	Output  interface{}
+	Outputs []interface{}
```

Comment:
Yeah, I started with that and it broke a fair number of worker tests that construct `StubResource`s directly rather than via `NewStubResources`. Fixing them required sprinkling awkward `[]interface{}{blah}`s around, so I thought I'd localise the awkwardness here instead. I'm alright with changing it back though.

Reply:
Fair enough. Maybe add a TODO that this could be improved? Maybe there's a helper function or something that could be done? Maybe land as is and consider a followup PR?

Reply:
Yeah, helper function is the right answer I think - I'll do that.